torchtext/experimental/datasets/raw/language_modeling.py

```diff
@@ -0,0 +1,146 @@
+import torch
+import logging
+import io
+from torchtext.utils import download_from_url, extract_archive
+
+URLS = {
+    'WikiText2':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip',
+    'WikiText103':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-103-v1.zip',
+    'PennTreebank':
+        ['https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.train.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.test.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.valid.txt']
+}
+
+
+class RawTextIterableDataset(torch.utils.data.IterableDataset):
+    """Defines an abstraction for raw text iterable datasets.
+    """
+
+    def __init__(self, iterator, start=0, num_lines=None):
+        """Initiate language modeling dataset.
+        """
+        super(RawTextIterableDataset, self).__init__()
+        self._iterator = iterator
+        self.start = start
+        self.num_lines = num_lines
+
```

Comment:
You can keep `setup_iter` there as text classification datasets as [here](https://github.com/pytorch/text/blob/9bdfb046c37b473accb7749ecc7c8e0e4ed7860f/torchtext/experimental/datasets/raw/text_classification.py#L47) and we will remove them all together once we agree with the abstraction.

Reply:
`setup_iter` restored in 321871d